File: Java-Core/src/main/java/presto/csharp/CSharpNativeCategoryBinding.java
package presto.csharp;

import presto.grammar.NativeCategoryBinding;
import presto.utils.CodeWriter;

public class CSharpNativeCategoryBinding extends NativeCategoryBinding {

	CSharpIdentifierExpression expression;
	
	public CSharpNativeCategoryBinding(CSharpIdentifierExpression expression) {
		this.expression = expression;
	}

	@Override
	public void toDialect(CodeWriter writer) {
		writer.append("C#: ");
		expression.toDialect(writer);
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/csharp/CSharpNativeCategoryBinding.js
var NativeCategoryBinding = require("./../grammar/NativeCategoryBinding").NativeCategoryBinding;

function CSharpNativeCategoryBinding(expression) {
	NativeCategoryBinding.call(this);
	this.expression = expression;
	return this;
}

CSharpNativeCategoryBinding.prototype = Object.create(NativeCategoryBinding.prototype);
CSharpNativeCategoryBinding.prototype.constructor = CSharpNativeCategoryBinding;

CSharpNativeCategoryBinding.prototype.toDialect = function(writer) {
    writer.append("C#: ");
    this.expression.toDialect(writer);
};

exports.CSharpNativeCategoryBinding = CSharpNativeCategoryBinding;
